packages/salesforcedx-lwc-language-server/src/baseIndexer.ts

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (c) 2025, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import { IFileSystemProvider } from '@salesforce/salesforcedx-lightning-lsp-common';
+import * as path from 'node:path';
+
+// Utility function to resolve workspace root
+export const getWorkspaceRoot = (workspaceRoot: string): string => path.resolve(workspaceRoot);
+
+// Utility function to get SFDX configuration
+export const getSfdxConfig = async (root: string, fileSystemProvider: IFileSystemProvider): Promise<any> => {
```

Comment:
not for this PR, but I think the client should be responsible for doing this, and then providing all the config the LS needs to do its job.